From: Ganapatrao Kulkarni <gkulkarni@caviumnetworks.com>
To: <linux-arm-kernel@lists.infradead.org>, <linux-kernel@vger.kernel.org>,
 <linux-pci@vger.kernel.org>, <Will.Deacon@arm.com>,
 <catalin.marinas@arm.com>, <bhelgaas@google.com>, <hanjun.guo@linaro.org>,
 <rrichter@cavium.com>, <Prasun.Kapoor@caviumnetworks.com>
Subject: [PATCH] arm64, pci, numa: Adding helper functions as required by pci
Date: Tue, 22 Dec 2015 16:08:50 +0530
Message-ID: <1450780730-13667-1-git-send-email-gkulkarni@caviumnetworks.com>
Patch-mainline: No yet. It is discussed if this will go in numa v9 or stay as it is

Adding helper functions and necessary code to make
pci driver(pci-host-generic) numa aware.

This patch is on top of arm64-numa v7.
http://www.spinics.net/lists/arm-kernel/msg460813.html

Signed-off-by: Ganapatrao Kulkarni <gkulkarni@caviumnetworks.com>
Signed-off-by: Matthias Brugger	<mbrugger@suse.com>
---
 drivers/pci/host/pci-host-generic.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/pci/host/pci-host-generic.c
+++ b/drivers/pci/host/pci-host-generic.c
@@ -317,6 +317,7 @@ static int gen_pci_probe(struct platform
 	of_pci_check_probe_only();
 
 	of_id = of_match_node(gen_pci_of_match, np);
+	set_dev_node(dev, of_node_to_nid(np));
 	pci->cfg.ops = (struct gen_pci_cfg_bus_ops *)of_id->data;
 	pci->host.dev.parent = dev;
 	INIT_LIST_HEAD(&pci->host.windows);
